src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -409,90 +409,58 @@ private static void RemoveToolLanguage(JObject run)
 
         private static void UpdateAllReportingDescriptorPropertyTypes(JObject run)
         {
-            // Access and modify run.tool
-            JObject tool = (JObject)run["tool"];
-            UpdateToolReportingDescriptorPropertyTypes(tool);
-
-            // Access and modify run.conversion.tool
-            if (run["conversion"] is JObject conversion)
+            string[] reportingDescriptorPathsToUpdate =
             {
-                tool = (JObject)conversion["tool"];
-                UpdateToolReportingDescriptorPropertyTypes(tool);
-            }
-        }
+                "tool.driver.notificationDescriptors[]",
+                "conversion.tool.driver.notificationDescriptors[]",
+                "tool.extensions[].notificationDescriptors[]",
+                "conversion.tool.extensions[].notificationDescriptors[]",
+                "tool.driver.ruleDescriptors[]",
+                "conversion.tool.driver.ruleDescriptors[]",
+                "tool.extensions[].ruleDescriptors",
+                "conversion.tool.extensions[].ruleDescriptors[]"
+            };
 
-        private static void UpdateToolReportingDescriptorPropertyTypes(JObject tool)
-        {
-            // Access and modify tool.driver
-            if (tool["driver"] is JObject driver)
-            {
-                UpdateToolComponentReportingDescriptorPropertyTypes(driver);
-            }
+            ApplyActionOnNodeIfPathExists(reportingDescriptorPathsToUpdate, run, UpdateReportingDescriptorPropertyTypes);
```

Comment:
>ApplyActionOnNodeIfPathExists [](start = 12, length = 29)

am i over optimizing here? unsure. Maybe a simpler approach would be to have the actual node modification logic in this method itself. at top level caller level (updateTC..) we will have only one single method call which does the job. i like the thought that any new change of this sort would introduce only one new method, not two.